src/python/genny/genny_auto_tasks.py

```diff
@@ -0,0 +1,125 @@
+"""
+Output json that can be used as input to evergreen's generate.tasks, representing genny workloads to be run
+"""
+import argparse
+import os
+import re
+import sys
+import subprocess
+import textwrap
+
+from shrub.config import Configuration
+from shrub.command import CommandDefinition
+from shrub.variant import TaskSpec
+
+
+def to_snake_case(str):
+    """
+    Converts str to snake_case, useful for generating test id's
+    From: https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
+    :return: snake_case version of str.
+    """
+    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', str)
+    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
+
+
+def modified_workload_files():
+    """
+    Returns a list of filenames for workloads that have been modified according to git, relative to origin/master.
+    :return: a list of filenames in the format subdirectory/Task.yml
+    """
+    # Return the names of files in ../workloads/ that have been added/modified/renamed since the common ancestor of HEAD and origin/master
+    try:
+        out = subprocess.check_output(
+            'git diff --name-only --diff-filter=AMR $(git merge-base HEAD origin/master) -- ../workloads/', shell=True)
+    except subprocess.CalledProcessError as e:
+        sys.exit(e.output)
```

Comment:
in general `sys.exit` should only be in the "main" function otherwise other code will be unable to call this code without fear of it taking down the whole process - let's `print(e.output)` and then `raise e` which will still cause us to abort the whole process.

Reply:
Ah, good to know, thanks. Made the change